src/extension/cli/server.ts

```diff
@@ -0,0 +1,86 @@
+
+import { ChildProcessWithoutNullStreams, spawn } from 'node:child_process'
+import path from 'node:path'
+import fs from 'node:fs/promises'
+
+import vscode, { Disposable } from 'vscode'
+
+import { SERVER_ADDRESS } from '../../constants'
+
+import ServerError from './serverError'
+
+export interface IServerConfig {
+    assignPortDynamically?: boolean
+    retryOnFailure?: boolean
+    maxNumberOfIntents: number
+}
+
+class Server implements Disposable {
```

Comment:
I feel like the name `Server` is extremely vague. How would you feel about something like `RunmeServer`?

Reply:
Totally open to that, however, just wanted to echo what I said before: lets not use the term `CLI` anywhere here since Runme's Command Line Interface is not involved whatsoever.

Reply:
Changed as suggested.